### PR TITLE
Sklearn Liniar Regression VS Custom Liniar Regression

### PR DESCRIPTION
**Pull request by GabrielVrabie007**
Trained a model for cars price prediction using Liniar Regression from Scikit-Learn library and my custom Liniar Regression.
In the first part I trained model using all columns but in second part I used only columns that has correlation to target column between 0,5 to 0.8. 
@danielavornic

### CODE REVIEW

**Review by danielavornic — APPROVED** [review]
About the Custom LR Error:
La modul cum functioneaza gradient descent, weights-urile in implementarea data cresc exponential, "explodeaza" si ajung sa faca overflow/underflow. Uneori e mai explicita eroarea data, alteori not really, precum in exemplul dat.  Eu am incercat sa setez learning rate ceva gen 0.0000000000001, si asa nu am eroare, dar scorul e foarte mic. The issue seems to be bc of the values in some columns (some are in the range of thousands, other millions, others are super low) from the given data. This variation in scales is causing numerical instability in the gradient descent calculations, since there are multiplications there. I see for the selected features, the custom algorithm is working fine, that’s because you eliminated the “problematic” columns, in the sense of numeric values, from the data. Sklearn uses another algorithm - Ordinary least squares, and some other optimizations, so that’s why it’s not an issue for it. 
There is a future chapter dedicated to Feature Scaling which might address situations like this, it will discuss normalization/standardization.

Overall you did great :) 
You followed the requirements, the comments and explanations were good, and feature encoding was a nice touch.
